packages/twenty-server/src/modules/messaging/message-import-manager/drivers/imap/services/imap-find-sent-folder.service.ts

```diff
@@ -0,0 +1,167 @@
+      }
+
+      const candidateSentFolder = await this.findSentFolderByCandidates(
+        client,
+        list,
+      );
+
+      if (candidateSentFolder) {
+        return candidateSentFolder;
+      }
+
+      this.logger.warn(
+        'No sent folder found. Only inbox messages will be imported.',
+      );
+
+      return null;
+    } catch (error) {
+      this.logger.warn(`Error listing folders: ${error.message}`);
+
+      return null;
+    }
+  }
+
+  private async findSentFolderBySpecialUse(
+    client: ImapFlow,
+    list: ListResponse[],
+  ): Promise<string | null> {
+    for (const folder of list) {
+      if (folder.specialUse && folder.specialUse.includes('\\Sent')) {
+        this.logger.log(
+          `Found sent folder via special-use flag: ${folder.path}`,
+        );
+
+        const hasMessages = await this.validateFolderHasMessages(
+          client,
+          folder.path,
+        );
+
+        if (hasMessages) {
+          return folder.path;
+        }
+
+        this.logger.warn(
+          `Special-use sent folder "${folder.path}" is empty, checking other candidates`,
+        );
+
+        break;
+      }
+    }
+
+    return null;
+  }
+
+  private async findSentFolderByCandidates(
+    client: ImapFlow,
+    list: ListResponse[],
+  ): Promise<string | null> {
+    const candidateFolders = this.getSentFolderCandidatesByRegex(list);
+
+    for (const folder of candidateFolders) {
+      const hasMessages = await this.validateFolderHasMessages(client, folder);
+
+      if (hasMessages) {
+        this.logger.log(`Selected sent folder via pattern match: ${folder}`);
+
+        return folder;
+      }
+    }
+
+    if (candidateFolders.length > 0) {
+      this.logger.log(
+        `Using first candidate sent folder: ${candidateFolders[0]} (no messages found in any candidate)`,
+      );
+
+      return candidateFolders[0];
+    }
+
+    return null;
+  }
+
+  private getSentFolderCandidatesByRegex(list: ListResponse[]): string[] {
+    // Comprehensive regex pattern for legacy IMAP servers
+    // Source: https://imapsync.lamiral.info/FAQ.d/FAQ.Folders_Mapping.txt
+    // Based on imapsync's regextrans2 examples (originally "Sent|Sent Messages|Gesendet")
+    // Extended with additional common localizations for broader provider/language support
+    const sentFolderPattern =
+      /^(.*\/)?(sent|sent[\s_-]?(items|mail|messages|elements)?|envoy[éê]s?|[ée]l[ée]ments[\s_-]?envoy[éê]s|gesendet|gesendete[\s_-]?elemente|enviados?|elementos[\s_-]?enviados|itens[\s_-]?enviados|posta[\s_-]?inviata|inviati|보낸편지함|\[gmail\]\/sent[\s_-]?mail)$/i;
+
+    const candidateFolders = [];
+
+    for (const folder of list) {
+      if (sentFolderPattern.test(folder.path)) {
+        this.logger.debug(
+          `Found potential sent folder via pattern match: ${folder.path}`,
+        );
+        candidateFolders.push(folder.path);
+      }
+    }
+
+    return candidateFolders;
+  }
+
+  private async validateFolderHasMessages(
```

Comment:
doesFolderHaveMessages (we could actually return the folderMessageCount has we have it?)